static/scenes/railroad/js/game.js

```diff
@@ -34,14 +43,42 @@ export class Game {
       this.renderer.setSize(window.innerWidth, window.innerHeight);
     });
 
+    this.raycaster = new THREE.Raycaster();
+    this.scoreboard = new app.shared.Scoreboard(this, undefined, app.Constants.NUM_LEVELS);
+
     // TODO: Put this in a level initialization section.
-    this.placeholderScene = new PlaceholderScene();
-    this.cameraSystem = new CameraSystem(this.camera, this.placeholderScene);
+    this.placeholderScene = new app.PlaceholderScene();
+    this.cameraSystem = new app.CameraSystem(this.camera, this.placeholderScene);
 
+    this.setUpListeners();
     this.mainLoop();
   }
 
+  pause() {
+    this.paused = true;
+    this.previousSeconds = null;
+  }
+
+  resume() {
+    this.paused = false;
+    this.previousSeconds = Date.now() / 1000;
+    this.mainLoop();
+  }
+
+  restart() {
+    console.log('TODO');
+    this.paused = false;
+    this.mainLoop();
```

Comment:
Would be nice to tweak the mainLoop function somehow so that it can't have two requestAnimationFrame loops running at the same time. If restart gets called while the game is running it might accidentally double up.

One possible way: Have a variable that tracks whether the game is running (instead of paused) and then have a startMainLoop() function that checks that the game isn't running before calling mainLoop

Reply:
This seemed like a larger refactor, so I'll leave that for a future change. I'd like to clarify the constraints for the restart callback. I wonder if it's okay verify the game is paused before proceeding? Assuming this works, I've at least protected against starting the second run loop for now.